### PR TITLE
Test Spark 3.3.0-rc3

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

https://www.mail-archive.com/dev@spark.apache.org/msg29048.html

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by cfmcgrady** [comment]
The failed UT is caused by https://github.com/apache/spark/pull/36591

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/spark/kyuubi/SparkSQLEngineDeregisterSuite.scala [hunk 1]
```diff
@@ -75,7 +75,12 @@ class SparkSQLEngineDeregisterMsgSuite extends SparkSQLEngineDeregisterSuite {
       // Upper case SQL types in error messages
       // see https://issues.apache.org/jira/browse/SPARK-39007
       // Use double quotes for SQL configs in error messages
-      "to int causes overflow,to \"INT\" causes overflow")
+      // see https://issues.apache.org/jira/browse/SPARK-39214
+      // Improve errors related to CAST
+      """to int causes overflow,
+        |to "INT" causes overflow,
```
Comment (cfmcgrady) [inline externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/spark/kyuubi/SparkSQLEngineDeregisterSuite.scala:81]: We don't need this error message `to "INT" causes overflow` anymore, the behavior of uppercasing SQL types in error messages is something that happened after Spark-3.3.

Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/spark/kyuubi/SparkSQLEngineDeregisterSuite.scala]: I simplify to just check "overflow"

**Review by cfmcgrady — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**